ateam_ai/src/trajectory_generation/trapezoidal_motion_profile.cpp

```diff
@@ -0,0 +1,360 @@
+    // https://www.wolframalpha.com/input?i=solve+d+%3D+m*+%28m-+s%29+%2F+a%2B+s+*+%28m-+s%29+%2F+a%2B+m*+%28m-+e%29+%2F+a+%2B+e*+%28m-+e%29+%2F+a+for+m
+
+    // max = max vel
+    // svel = start vel
+    // evel = end vel
+    // d = target_distance
+
+    // ts = (max - svel) / max_accel
+    // te = (max - evel) / max_accel
+
+    // avgs = (max + svel) * ts
+    // avge = (max + evel) * te
+
+    // d = avgs + avge
+    // d = (max + svel) * ts + (max + evel) * te
+    // d = max * ts + svel * ts + max * te + evel * te
+    // d = max * (max - svel) / max_accel + svel * (max - svel) / max_accel +
+    //     max * (max - evel) / max_accel + evel * (max - evel) / max_accel
+    // d * max_accel = max * (max - svel) + svel * (max - svel) +
+    //                 max * (max - evel) + evel * (max - evel)
+    // d * max_accel = max^2 - max * svel + svel * max - svel^2 +
+    //                 max^2 - max * evel + evel * max - evel^2
+    // d * max_accel + svel^2 + evel^2 = 2 * max^2
+    // (d * max_accel + svel^2 + evel^2) / 2 = max^2
+    // sqrt((d * max_accel + svel^2 + evel^2) / 2) = +-max
+
+    // Don't allow negative speed so we can just assume positive is only correct value
+    double nonlimit_max_vel =
+      sqrt((target_dist * max_accel + start_vel * start_vel + end_vel * end_vel) / 2);
+
+    // Phase 1 is start_vel to nonlimit_max_vel
+    const AccelerationPhaseInputs phase_1_to_nonlimit_vel_input{
+      .start_vel = start_vel,
+      .end_vel = nonlimit_max_vel,
+      .accel_limit = max_accel};
+
+    // Phase 3 is max_accel to end_vel
+    const AccelerationPhaseInputs phase_3_from_nonlimit_vel_input{
+      .start_vel = nonlimit_max_vel,
+      .end_vel = end_vel,
+      .accel_limit = max_accel};
+
+    const AccelerationPhaseResults phase_1_to_nonlimit_vel = get_acceleration_phase_results(
+      phase_1_to_nonlimit_vel_input);
+    const AccelerationPhaseResults phase_3_from_nonlimit_vel = get_acceleration_phase_results(
+      phase_3_from_nonlimit_vel_input);
+
+
+    // Skip phase 2
+    trapezoidal_coeffs.phase_2.start_time = phase_1_to_nonlimit_vel.time_required;
+    trapezoidal_coeffs.phase_2.start_dist = phase_1_to_nonlimit_vel.distance_required;
+    trapezoidal_coeffs.phase_2.start_vel = nonlimit_max_vel;
+    trapezoidal_coeffs.phase_2.duration = 0;
+    trapezoidal_coeffs.phase_2.end_vel = nonlimit_max_vel;
+
+    trapezoidal_coeffs.phase_3.start_time = trapezoidal_coeffs.phase_2.start_time;
+    trapezoidal_coeffs.phase_3.start_dist = trapezoidal_coeffs.phase_2.start_dist;
+    trapezoidal_coeffs.phase_3.start_vel = nonlimit_max_vel;
+    trapezoidal_coeffs.phase_3.duration = phase_3_from_nonlimit_vel.time_required;
+    trapezoidal_coeffs.phase_3.end_vel = end_vel;
+  }
+
+  Trajectory1d output = sample_trapezoidal_profile(trapezoidal_coeffs, max_accel, dt);
+
+  // Add offset start pos
+  for (auto & sample : output.samples) {
+    sample.pos += start_pos;
+  }
```

Comment:
Consider [std::accumulate](https://en.cppreference.com/w/cpp/algorithm/accumulate)

```suggestion
  sample.pos = std::accumulate(output.samples.begin(), output.samples.end(), sample.pos);
```

Reply:
I'm not sure `std::accumulate` is the right one, this is adding the start position offset to all samples. I think this would be the way to go if  we wanted to use `<algorithm>`
```
std::transform(output.samples.begin(),
               output.samples.end(),
               output.samples.begin(),
               [start_pus](double p) {
                 return p + start_pos;
               });
```

Reply:
Actually, I'll just swap to that since it's not clear with just a glance